mlos_bench/mlos_bench/launcher.py

```diff
@@ -237,9 +251,14 @@ def _expand_vars(self, value: Any) -> Any:
         NOTE: `self.global_config` must be set.
         """
         if isinstance(value, str):
+            if "$" in value:
+                _LOG.warning("Expand vars: %s", value)
             return Template(value).safe_substitute(self.global_config)
         if isinstance(value, dict):
-            return {key: self._expand_vars(val) for (key, val) in value.items()}
+            # Note: we use a loop instead of dict comprehension in order to
+            # allow secondary expansion of subsequent values immediately.
+            for (key, val) in value.items():
+                value[key] = self._expand_vars(val)
```

Comment:
that's
```suggestion
            value = {key: self._expand_vars(val) for (key, val) in value.items()}
```

Reply:
oh wait never mind, that won't work if there is a chain of dependencies

Reply:
Exactly, hence the change/comment.  I'm still not sure this is the best way, but it accomplishes the desired behavior we'd discussed in a previous PR as demonstrated in the tests we're adding here.